src/main/java/com/nvs/store/service/CartService.java

```diff
@@ -0,0 +1,202 @@
+    private final CartItemRepository cartItemRepository;
+    private final UserRepository userRepository;
+    private final ProductRepository productRepository;
+
+    public CartDTO findActualCartByEmail() {
+        String userEmail = getUserEmailFromContext();
+        User user = userRepository
+                .findByEmail(userEmail)
+                .orElseThrow(() -> new RuntimeException(
+                        String.format("User with email [%s] not found", userEmail))
+                );
+        Cart cart = cartRepository
+                .findByUserIdAndActualTrue(user.getId());
+        CartDTO cartDTO;
+        if (cart == null) {
+            Cart newCart = Cart.builder()
+                    .user(user)
+                    .actual(true)
+                    .cartItems(List.of())
+                    .build();
+            cartRepository.save(newCart);
+            cartDTO = CartUtils.convertToCartDto.apply(newCart);
+            return cartDTO;
+        }
+        cartDTO = CartUtils.convertToCartDto.apply(cart);
+        return cartDTO;
+    }
+
+    public List<CartDTO> getAllCarts() {
+        String userEmail = getUserEmailFromContext();
+        User user = userRepository
+                .findByEmail(userEmail)
+                .orElseThrow(() -> new RuntimeException(
+                        String.format("User with email [%s] not found", userEmail))
+                );
+        return cartRepository
+                .findByUserId(user.getId())
+                .stream()
+                .map(CartUtils.convertToCartDto)
+                .toList();
+    }
+
+    public CartDTO addProductToCart(CartItemRequest cartItemRequest) {
+        String userEmail = getUserEmailFromContext();
+        User user = userRepository.findByEmail(userEmail)
+                .orElseThrow(() -> new RuntimeException(
+                        String.format("User with username %s not found", userEmail))
+                );
+
+        Product product = productRepository
+                .findById(cartItemRequest
+                        .getProductId())
+                .orElseThrow(() ->
+                        new RuntimeException(
+                                String.format("Product with id [%d] does not exist", cartItemRequest.getProductId()))
+                );
+        Cart cart = cartRepository
+                .findByUserIdAndActualTrue(user.getId());
+
+        List<CartItem> cartItems = cartItemRepository.findCartItemByCart(cart);
+        for (CartItem cartItem : cartItems) {
+            if (cartItem.getProduct().getId().equals(product.getId())) {
+                int newQuantity = cartItem.getQuantity() + cartItemRequest.getQuantity();
+                validateAvailableProductQuantities(product, newQuantity);
+                cartItem.setQuantity(newQuantity);
+                cartItemRepository.save(cartItem);
+                return CartUtils.convertToCartDto.apply(cartRepository.findByUserIdAndActualTrue(user.getId()));
```

Comment:
its not necessary to return cartdto on addProductToCart action, you can return just updated cartItem details

Reply:
done ✅